### PR TITLE
Upgrade Pad op to opset3

### PR DESCRIPTION
**Pull request by bani-intelaipg**
Use NGraph Opset3 for several Pad ops

### CODE REVIEW

File: ngraph_bridge/ngraph_builder.cc [hunk 1]
```diff
@@ -2564,26 +2565,129 @@ static Status TranslatePadOp(const Node* op,
         "elements");
   }
 
-  ng::CoordinateDiff padding_below(paddings.size() / 2);
-  ng::CoordinateDiff padding_above(paddings.size() / 2);
-  ng::Shape padding_interior(paddings.size() / 2);
+  ng::CoordinateDiff padding_before(paddings.size() / 2);
```
Comment (adk9) [inline ngraph_bridge/ngraph_builder.cc:2568]: Minor style change: `pad_begin`/`pad_end` for consistency?

File: ngraph_bridge/ngraph_builder.cc [hunk 2]
```diff
@@ -2564,26 +2565,129 @@ static Status TranslatePadOp(const Node* op,
...
+  auto pads_begin_node = make_shared<ng::opset3::Constant>(
+      ng::element::i64, ng::Shape{padding_before.size()}, padding_before);
+  auto pads_end_node = make_shared<ng::opset3::Constant>(
+      ng::element::i64, ng::Shape{padding_after.size()}, padding_after);
 
   // For PadV1 it seems the value is always zero.
-  auto pad_val_op = ConstructNgNode<ng::op::Constant>(
+  auto pad_val_op = ConstructNgNode<ng::opset3::Constant>(
+      op->name(), ng_input->get_element_type(), ng::Shape{},
+      std::vector<std::string>{"0"});
```
Comment (adk9) [inline ngraph_bridge/ngraph_builder.cc:2588]:
```
  auto pad_val_op = ConstructNgNode<ng::opset3::Constant>(
      op->name(), ng_input->get_element_type(), ng::Shape{}, {0});
```

File: ngraph_bridge/ngraph_builder.cc [hunk 3]
```diff
@@ -2564,26 +2565,129 @@ static Status TranslatePadOp(const Node* op,
...
+                                   " is not an allowed padding mode.");
+  }
+
+  if (paddings.size() % 2 != 0) {
+    return errors::InvalidArgument(
+        "Constant node for paddings does not have an even number of "
+        "elements");
+  }
+
+  ng::CoordinateDiff padding_before(paddings.size() / 2);
```
Comment (adk9) [inline ngraph_bridge/ngraph_builder.cc:2670]: You can put this common part into a helper function to construct `opset3::Pad` that all 3 pad ops can use.

**Comment by bani-intelaipg** [comment]
The Grappler build is intermittently breaking on Tf Pytests (SumReductionTest.testFloat32), which we will debug separately. This PR is now ready to  be merged to master.

File: ngraph_bridge/ngraph_builder.cc [hunk 4]
```diff
@@ -2547,48 +2547,93 @@ static Status TranslatePackOp(const Node* op, const std::vector<const Tensor*>&,
...
                              const std::vector<const Tensor*>& static_input_map,
                              Builder::OpMap& ng_op_map) {
-  shared_ptr<ng::Node> ng_input, ng_paddings_op;
-  TF_RETURN_IF_ERROR(GetInputNodes(ng_op_map, op, &ng_input, &ng_paddings_op));
+  shared_ptr<ng::Node> ng_input, ng_paddings_op, pad_val_op, result_pad_op;
 
-  std::vector<int64> paddings;
-  TF_RETURN_IF_ERROR(GetStaticInputVector(op, 1, static_input_map, &paddings));
+  // Set inputs and pad_val_op
+  if (op->type_string() == "Pad" || op->type_string() == "MirrorPad") {
```
Comment (adk9) [inline ngraph_bridge/ngraph_builder.cc:2557]: Nice. Now that you've a common helper for all 3 cases, you could just rename it to `TranslatePadOp` and use it directly in `TRANSLATE_OP_MAP` for all 3 pad ops?

File: ngraph_bridge/ngraph_builder.cc (hunk 2 above)
Comment (bani-intelaipg) [reply ngraph_bridge/ngraph_builder.cc]: Oops, i missed that. Sure.

File: ngraph_bridge/ngraph_builder.cc (hunk 4 above)
Comment (bani-intelaipg) [reply ngraph_bridge/ngraph_builder.cc:2557]: Oops, i missed that. Sure.

**Pull request merged**